internal/scorer/config_test.go

```diff
@@ -0,0 +1,214 @@
+package scorer
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/ossf/criticality_score/internal/scorer/algorithm"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestInput_ToAlgorithmInput(t *testing.T) {
+	type fields struct {
+		Bounds       *algorithm.Bounds
+		Condition    *Condition
+		Field        string
+		Distribution string
+		Tags         []string
+		Weight       float64
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *algorithm.Input
+		wantErr bool
+	}{
+		{
+			name: "unknown distribution error",
+			fields: fields{
+				Field: "test",
+			},
+			want:    &algorithm.Input{},
+			wantErr: true,
+		},
+		{
+			name: "distribution value set",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+			},
+			want: &algorithm.Input{
+				Source:       algorithm.Value(algorithm.Field("test")),
+				Distribution: algorithm.LookupDistribution("linear"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Valid condition",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+				Condition: &Condition{
+					FieldExists: "test",
+				},
+			},
+			want: &algorithm.Input{
+				Distribution: algorithm.LookupDistribution("linear"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Invalid condition",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+				Condition: &Condition{
+					FieldExists: "test",
+					Not: &Condition{
+						FieldExists: "test",
+					},
+				},
+			},
+			want:    &algorithm.Input{},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			i := &Input{
+				Bounds:       tt.fields.Bounds,
+				Condition:    tt.fields.Condition,
+				Field:        tt.fields.Field,
+				Distribution: tt.fields.Distribution,
+				Tags:         tt.fields.Tags,
+				Weight:       tt.fields.Weight,
+			}
+			got, err := i.ToAlgorithmInput()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ToAlgorithmInput() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				return
+			}
+
+			// Comparing specific fields independently because some of the structs have a func as a field which
+			// can't be used for comparison with reflect.DeepEqual()
+
+			if got.Distribution.String() != tt.want.Distribution.String() {
+				t.Errorf("ToAlgorithmInput() got = %v, want %v", got, tt.want)
+			}
+			if got.Bounds != tt.want.Bounds {
+				t.Errorf("ToAlgorithmInput() got = %v, want %v", got, tt.want)
+			}
+			if got.Weight != tt.want.Weight {
+				t.Errorf("ToAlgorithmInput() got = %v, want %v", got, tt.want)
+			}
+			if !reflect.DeepEqual(got.Tags, tt.want.Tags) {
+				t.Errorf("ToAlgorithmInput() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestLoadConfig(t *testing.T) {
+	type args struct {
+		r string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *Config
+		wantErr bool
+	}{
```

Comment:
Might be worth adding a case where the yaml is valid, but the config is not a valid scorer config (i.e. yaml parsing succeeds, but turning it into a Scorer fails).

Reply:
That condition is already covered in the test cases. If you see something off, please let me know.